dotnet/src/Agents/Core/AgentGroupChat.cs

```diff
@@ -67,10 +71,25 @@ public async override IAsyncEnumerable<ChatMessageContent> InvokeAsync([Enumerat
             this.IsComplete = false;
         }
 
+        this.Logger.LogDebug("[{MethodName}] Invoking chat: {Agents}", nameof(InvokeAsync), string.Join(", ", this.Agents.Select(a => $"{a.GetType()}:{a.Id}")));
+
         for (int index = 0; index < this.ExecutionSettings.TerminationStrategy.MaximumIterations; index++)
         {
             // Identify next agent using strategy
-            Agent agent = await this.ExecutionSettings.SelectionStrategy.NextAsync(this.Agents, this.History, cancellationToken).ConfigureAwait(false);
+            this.Logger.LogDebug("[{MethodName}] Selecting agent: {StrategyType}", nameof(InvokeAsync), this.ExecutionSettings.SelectionStrategy.GetType());
+
+            Agent agent;
+            try
+            {
+                agent = await this.ExecutionSettings.SelectionStrategy.NextAsync(this.Agents, this.History, cancellationToken).ConfigureAwait(false);
+            }
+            catch
+            {
+                this.Logger.LogError("[{MethodName}] Unable to determine next agent.", nameof(InvokeAsync));
```

Comment:
We can also pass `exception` instance to `LogError` here.

Reply:
Creates build warning / pipeline failure

Reply:
> Creates build warning / pipeline failure

Which exactly? I don't have any warnings locally:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/52181eb0-9414-453f-982c-b935683cf2de)

Reply:
Your original suggestion:

`this.Logger.LogError(exception, exception.Message);`

Reply:
Updated